engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnVariableMappingBehavior.java

```diff
@@ -119,14 +125,39 @@ public Either<Failure, Void> applyOutputMappings(
       variableBehavior.mergeDocument(
           elementInstanceKey, processDefinitionKey, processInstanceKey, bpmnProcessId, variables);
     } else if (isConnectedToEventBasedGateway(element)
-        || (element.getElementType() == BpmnElementType.BOUNDARY_EVENT && !isErrorEvent(element))
+        || ((element.getElementType() == BpmnElementType.BOUNDARY_EVENT) && !isErrorEvent(element))
         || element.getElementType() == BpmnElementType.START_EVENT) {
       // event variables are set local variables instead of temporary variables
       final var localVariables = variablesState.getVariablesLocalAsDocument(elementInstanceKey);
       variableBehavior.mergeDocument(
           scopeKey, processDefinitionKey, processInstanceKey, bpmnProcessId, localVariables);
     }
 
+    if (element.getElementType() == BpmnElementType.BOUNDARY_EVENT && isErrorEvent(element)) {
+      // extract error code, error message to variables
+      final List<DirectBuffer> variableNames = new ArrayList<>();
+      final ExecutableBoundaryEvent boundaryEvent = (ExecutableBoundaryEvent) element;
+      final ExecutableError error = boundaryEvent.getError();
+      error
+          .getErrorMessageVariable()
+          .filter(StringUtils::isNotBlank)
+          .map(BufferUtil::wrapString)
+          .ifPresent(variableNames::add);
+
+      error
+          .getErrorCodeVariable()
+          .filter(StringUtils::isNotBlank)
+          .map(BufferUtil::wrapString)
+          .ifPresent(variableNames::add);
+
+      if (!variableNames.isEmpty()) {
+        final var localVariables =
+            variablesState.getVariablesAsDocument(elementInstanceKey, variableNames);
+        variableBehavior.mergeDocument(
+            scopeKey, processDefinitionKey, processInstanceKey, bpmnProcessId, localVariables);
+      }
+    }
```

Comment:
❌ These changes are unnecessary because the error code and error message variables are already added to the event trigger's variables, so the variables are created in the process automatically.